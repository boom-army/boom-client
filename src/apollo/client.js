import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          users: offsetLimitPagination(),
        },
      },
    },
  }),
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
  },
  context: async ({ req }) => {
    const token = req?.headers?.authorization || null;
    const user= token ? JSON.parse(localStorage.getItem("user")) : null;
    return { user, isLoggedIn: !!token };
  },
});
