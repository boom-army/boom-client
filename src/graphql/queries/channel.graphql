mutation addChannel(
  $mintAuthority: String!
  $name: String!
  $family: String!
  $description: String
  $image: String
  $status: String
  $channelParentId: String
) {
  addChannel(
    mintAuthority: $mintAuthority
    name: $name
    family: $family
    description: $description
    image: $image
    status: $status
    channelParentId: $channelParentId
  ) {
    id
    mintAuthority
    name
    family
    description
    image
    channelParentId
    status
    membersCount {
      count
      avatars
    }
    createdAt
    updatedAt
  }
}

query channels {
  channels {
    id
    mintAuthority
    name
    family
    description
    image
    channelParentId
    status
    verified
    membersCount {
      count
      avatars
    }
  }
}

mutation channelUnlink($channelId: ID!) {
  channelUnlink(channelId: $channelId)
}

fragment TweetData on Tweet {
  id
  text
  tags
  isTweetMine
  commentsCount
  retweetsCount
  isRetweet
  tipsCount
  createdAt
  channel {
    id
  }
  parentTweet {
    id
    text
    user {
      id
      handle
      avatar
      data {
        avatarIsNFT
      }
    }
  }
  files {
    id
    url
  }
  gif {
    id
    title
    fixedHeightUrl
    originalUrl
  }
  nft {
    id
    publicKey
    name
    symbol
    description
    sellerFeeBasisPoints
    externalUrl
    image
    attributes {
      traitType
      value
    }
    collection {
      name
      family
    }
    properties {
      files {
        uri
        type
      }
      category
      creators {
        address
        share
      }
    }
  }
  user {
    id
    publicAddress
    avatar
    handle
    consumerName
    data {
      avatarIsNFT
    }
  }
  reactions {
    id
    emojiId
    skin
    isMine
    count
  }
}

query channelFeed($channelId: ID!, $offset: Int, $limit: Int) {
  channelFeed(channelId: $channelId, offset: $offset, limit: $limit) {
    ...TweetData
  }
}
