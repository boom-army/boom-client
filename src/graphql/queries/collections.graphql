query getCollection($name: String!) {
  getCollection(name: $name)
}
