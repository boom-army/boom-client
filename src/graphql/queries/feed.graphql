query feed($offset: Int!, $limit: Int, $global: Boolean) {
  feed(offset: $offset, limit: $limit, global: $global) {
    id
    text
    tags
    isTweetMine
    commentsCount
    retweetsCount
    isRetweet
    tipsCount
    createdAt
    parentTweet {
      id
    }
    files {
      id
      url
    }
    gif {
      id
      title
      fixedHeightUrl
      originalUrl
    }
    nft {
      id
      publicKey
      name
      symbol
      description
      sellerFeeBasisPoints
      externalUrl
      image
      attributes {
        traitType
        value
      }
      collection {
        name
        family
      }
      properties {
        files {
          uri
          type
        }
        category
        creators {
          address
          share
        }
      }
    }
    user {
      id
      publicAddress
      avatar
      handle
      consumerName
    }
    reactions {
      id
      emojiId
      skin
      isMine
      count
    }
  }
}
