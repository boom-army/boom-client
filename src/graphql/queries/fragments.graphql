fragment BaseUser on User {
  id
  avatar
  handle
  consumerName
  publicAddress
  data {
    avatarMint
    avatarUpdateAuthority
  }
  isTyping
}

fragment SlimTweet on Tweet {
  id
  text
  createdAt
  user {
    ...BaseUser
  }
}

fragment TweetData on Tweet {
  id
  text
  isTweetMine
  commentsCount
  retweetsCount
  isRetweet
  tipsCount
  createdAt
  tags {
    id
    text
  }
  channel {
    id
  }
  masterTweet {
    ...SlimTweet
  }
  parentTweet {
    ...SlimTweet
  }
  files {
    id
    url
  }
  gif {
    id
    title
    fixedHeightUrl
    originalUrl
  }
  nft {
    id
    publicKey
    name
    symbol
    description
    sellerFeeBasisPoints
    externalUrl
    image
    attributes {
      traitType
      value
    }
    collection {
      name
      family
    }
    properties {
      files {
        uri
        type
      }
      category
      creators {
        address
        share
      }
    }
  }
  user {
    ...BaseUser
  }
  reactions {
    id
    emojiId
    skin
    isMine
    count
  }
}

fragment ChannelData on Channel {
  id
  mintAuthority
  name
  family
  description
  image
  channelParentId
  status
  verified
  membersCount {
    count
    avatars
  }
}
