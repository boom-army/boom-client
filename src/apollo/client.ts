import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          heroFeed: offsetLimitPagination(),
          getChannelById: offsetLimitPagination(),
          users: offsetLimitPagination(),
          searchTweets: offsetLimitPagination(),
          searchUser: offsetLimitPagination(),
          mentions: offsetLimitPagination(),
        },
      },
    },
  }),
  credentials: 'include',
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
    "Access-Control-Allow-Origin" : "*",
  },
  defaultOptions: {
    watchQuery: {
      context: async ({ req }: any) => {
        const token = req?.headers?.authorization || null;
        const user = token ? localStorage.getItem("user")?.toString() : null;
        return { user, isLoggedIn: !!token };
      },
    },
  },
});
