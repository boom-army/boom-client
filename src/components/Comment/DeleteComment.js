import React from "react";
import { DELETE_COMMENT } from "../../queries/comment";
import { TWEET } from "../../queries/tweet";
import { TrashIcon } from "../Icons";
import { useMutation } from "@apollo/client";
import { useParams } from "react-router-dom";
import { useSnackbar } from "notistack";

const DeleteComment = ({ id }) => {
  const { tweetId } = useParams();
  const { enqueueSnackbar } = useSnackbar();

  const [deleteCommentMutation, { loading }] = useMutation(DELETE_COMMENT, {
    variables: { id },
    update: (cache, { data: { deleteComment } }) => {
      const { tweet } = cache.readQuery({
        query: TWEET,
        variables: { id: tweetId },
      });

      cache.writeQuery({
        query: TWEET,
        data: {
          tweet: {
            ...tweet,
            commentsCount: tweet.commentsCount - 1,
            comments: tweet.comments.filter(
              (comment) => comment.id !== deleteComment.id
            ),
          },
        },
      });
    },
  });

  const handleDeleteComment = async () => {
    await deleteCommentMutation();
    enqueueSnackbar("Your comment has been deleted", { variant: "success" });
  };

  return <TrashIcon loading={loading} onClick={handleDeleteComment} />;
};

export default DeleteComment;
