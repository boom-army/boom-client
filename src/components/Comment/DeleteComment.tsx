import React from "react";
import { DELETE_COMMENT } from "../../queries/comment";
import { TWEET } from "../../queries/tweet";
import { TrashIcon } from "../Icons";
import { useMutation } from "@apollo/client";
import { useParams } from "react-router-dom";
import { useSnackbar } from "../../contexts/snackbar";
import { Comment, Tweet } from "../../generated/graphql";

interface DeleteProps {
  id: string;
}

const DeleteComment = ({ id }: DeleteProps) => {
  const { tweetId } = useParams();
  const { enqueueSnackbar } = useSnackbar();

  const [deleteCommentMutation, { loading }] = useMutation(DELETE_COMMENT, {
    variables: { id },
    update: (cache, { data: { deleteComment } }) => {
      const { tweet }: any = cache.readQuery({
        query: TWEET,
        variables: { id: tweetId },
      });

      cache.writeQuery({
        query: TWEET,
        data: {
          tweet: {
            ...tweet,
            commentsCount: tweet.commentsCount - 1,
            comments: tweet.comments.filter(
              (comment: Comment) => comment.id !== deleteComment.id
            ),
          },
        },
      });
    },
  });

  const handleDeleteComment = async () => {
    await deleteCommentMutation();
    enqueueSnackbar("Your comment has been deleted", { variant: "success" });
  };

  return <TrashIcon loading={loading} onClick={handleDeleteComment} />;
};

export default DeleteComment;
