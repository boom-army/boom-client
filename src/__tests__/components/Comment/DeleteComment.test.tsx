import React from "react";
import { MemoryRouter } from "react-router-dom";
import { act, fireEvent, render } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { SnackbarProvider } from "notistack";
import { DELETE_COMMENT } from "../../../queries/comment";
import DeleteComment from "../../../components/Comment/DeleteComment";

const values = {
  handle: "shy-cloud-4965",
  bio: "full stack developer",
  consumerName: "shy-cloud-4965",
  coverPhoto: "/default-cover.png",
  dob: "d",
  location: "",
  website: "abcdfd",
};

export const __mocks__: any = [
  {
    request: {
      query: DELETE_COMMENT,
      variables: {
        id: "cl084xdy815761s0n5omoxoem0",
      },
    },
    result: {
      data: {},
    },
  },
];

describe("<DeleteComponent/>: ", () => {
  test("renders correctly when User have profile", async () => {
    let rendered;
    await act(async () => {
      rendered = render(
        <MockedProvider mocks={__mocks__} addTypename={false}>
          <SnackbarProvider>
            <DeleteComment id={"cl084xdy815761s0n5omoxoem0"} />
          </SnackbarProvider>
        </MockedProvider>,
        { wrapper: MemoryRouter }
      );
    });
    expect(rendered).toMatchSnapshot();
  });
});
