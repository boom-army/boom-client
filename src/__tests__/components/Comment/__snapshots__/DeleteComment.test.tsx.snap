// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteComponent/>:  renders correctly when User have profile 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        class="css-1csj31"
        loading=""
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M20.746 5.236h-3.75V4.25c0-1.24-1.01-2.25-2.25-2.25h-5.5c-1.24 0-2.25 1.01-2.25 2.25v.986h-3.75c-.414 0-.75.336-.75.75s.336.75.75.75h.368l1.583 13.262c.216 1.193 1.31 2.027 2.658 2.027h8.282c1.35 0 2.442-.834 2.664-2.072l1.577-13.217h.368c.414 0 .75-.336.75-.75s-.335-.75-.75-.75zM8.496 4.25c0-.413.337-.75.75-.75h5.5c.413 0 .75.337.75.75v.986h-7V4.25zm8.822 15.48c-.1.55-.664.795-1.18.795H7.854c-.517 0-1.083-.246-1.175-.75L5.126 6.735h13.74L17.32 19.732z"
          />
          <path
            d="M10 17.75c.414 0 .75-.336.75-.75v-7c0-.414-.336-.75-.75-.75s-.75.336-.75.75v7c0 .414.336.75.75.75zm4 0c.414 0 .75-.336.75-.75v-7c0-.414-.336-.75-.75-.75s-.75.336-.75.75v7c0 .414.336.75.75.75z"
          />
        </g>
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="css-1csj31"
      loading=""
      viewBox="0 0 24 24"
    >
      <g>
        <path
          d="M20.746 5.236h-3.75V4.25c0-1.24-1.01-2.25-2.25-2.25h-5.5c-1.24 0-2.25 1.01-2.25 2.25v.986h-3.75c-.414 0-.75.336-.75.75s.336.75.75.75h.368l1.583 13.262c.216 1.193 1.31 2.027 2.658 2.027h8.282c1.35 0 2.442-.834 2.664-2.072l1.577-13.217h.368c.414 0 .75-.336.75-.75s-.335-.75-.75-.75zM8.496 4.25c0-.413.337-.75.75-.75h5.5c.413 0 .75.337.75.75v.986h-7V4.25zm8.822 15.48c-.1.55-.664.795-1.18.795H7.854c-.517 0-1.083-.246-1.175-.75L5.126 6.735h13.74L17.32 19.732z"
        />
        <path
          d="M10 17.75c.414 0 .75-.336.75-.75v-7c0-.414-.336-.75-.75-.75s-.75.336-.75.75v7c0 .414.336.75.75.75zm4 0c.414 0 .75-.336.75-.75v-7c0-.414-.336-.75-.75-.75s-.75.336-.75.75v7c0 .414.336.75.75.75z"
        />
      </g>
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
